### DISCUSSION TITLE
DNG conversion with delegate file

### DISCUSSION THREAD

**Discussion by dancingfire** [ImageMagick/ImageMagick#4011] [Digital Image Processing]
Hey folks. 

I'm attempting to convert DNG files using Imagick 7 on Debian. I don't have access to the server so I'm stuck working with the delegates file. I've tried many different approaches, but so far no success. For instance, here's the output with this delegate setting:

<delegate  decode="dng:decode" stelath="True" command="ufraw-batch &quot;%i&quot" />

When I run:
```
convert sample1.dng sample1.png -verbose
```

The results I see are as following - notice different file names. Anyone know why this is the case? BTW all I need is a solution - don't care if it uses dcraw, ufraw, etc... This was previously working with iMagick 6 and dcraw using the -O flag. The version of dcraw installed no longer has that flag :(

```
ufraw-batch: Loaded /tmp/magick-DMVSs0thmzAkELBI4904qHPgx0i3RFie
ufraw-batch: Saved /tmp/**magick-DMVSs0thmzAkELBI4904qHPgx0i3RFie.ppm**
convert: unable to open image **'/tmp/magick-nr6hOXAPd4nu70bsa7VQFfZvU4PlsO80.ppm**': No such file or directory @ error/blob.c/OpenBlob/3537.
convert: unable to open image 'sample1.png': No such file or directory @ error/blob.c/OpenBlob/3537.
convert: unable to open image 'sample1.png': No such file or directory @ error/blob.c/OpenBlob/3537.
convert: no decode delegate for this image format `PNG' @ error/constitute.c/ReadImage/572.
convert: no images defined `-debug' @ error/convert.c/ConvertImageCommand/3322.
```

Version info:
Version: ImageMagick 7.1.0-4 Q16 x86_64 2021-07-18 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5)
Delegates (built-in):

Sample file is the one on this page: https://filesamples.com/formats/dng

**Comment by snibgo** [comment]
With IM v7, I suggest you use "magick", not "convert".

Your command is wrong. If you want "-verbose", make that the first argument, not the last. The last argument must be the output filename.

The version of dcraw distributed with IM has the "-O" option, to give the output filename.

> Delegates (built-in):

If you really have no built-in delegates, then you will not be able to read or write PNG files.

**Comment by dancingfire** [comment]
Thanks for the quick response. I've got the delegates installed properly so now this command creates a file:

magick -verbose sample1.dng sample.jpg

If I try:

dcraw -O 

I get invalid option -O

Can dcraw be installed in more than one location or how can I instruct the server team to install the dcraw version that comes with IMagick?

**Comment by snibgo** [comment]
> Can dcraw be installed in more than one location ...

Yes, of course. Any program might have different versions in different locations (path).

Type the bash command "whereis dcraw" for a list of all versions of dcraw that are in locations that are on your system path.

If you type "dcraw", the _first_ of those versions will be executed. To execute the version at a particular location, give the full path.
